src/utils.ts

```diff
@@ -21,6 +21,34 @@ export const mac = ios || /Mac/.test(platform);
 * Utility functions used in one or more files
 */
 
+/**
+ * @internal
+ * @returns a number, representing ReturnType<typeof requestAnimationFrame>
+ * Expose a scheduler for after react's render cycle is over. Some
+ * internal functions use it, and testing infrastructure may use it as well
+ * see stackoverflow.com/questions/26556436/react-after-render-code/28748160#28748160
+ * 
+ * - If the owner argument is passed, an inner timeout of 0 is set as 'pendingTimeout'
+ *   so that it can be cancelled by the owner later
+ * - If an extraDelay is passed, the inner timeout waits Xms after the render cycle
+ */
+export type afterDOMUpdateHandle = {
+  raf: number,
+  timeout: ReturnType<typeof setTimeout>
+};
+export function setAfterDOMUpdate(f: ()=>void, extraDelay?:number): afterDOMUpdateHandle {
+  let timeoutHandle: ReturnType<typeof setTimeout>;
+  return {
+    raf: window.requestAnimationFrame(() => timeoutHandle = setTimeout(f, extraDelay)),
+    timeout: timeoutHandle
+  };
```

Comment:
close. The `afterDOMUpdateHandle` returned by this function will always have an undefined timeout, because `timeoutHandle` doesn't get set until after this function returns. You need to box it.

Reply:
Confirming that in JS, boxing has to be handled by wrapping something in an object?